druid-shell/src/platform/gtk/dialog.rs

```diff
@@ -45,6 +54,15 @@ pub(crate) fn get_file_dialog_path(
 
     dialog.set_select_multiple(options.multi_selection);
 
+    if let Some(ref file_types) = options.allowed_types {
+        for f in file_types {
+            dialog.add_filter(&file_filter(f));
+        }
+    }
+    if let Some(ref default_file_type) = options.default_type {
```

Comment:
Let's remove the `ref` keywords and use `&options....` instead. `ref` is [a legacy pattern](https://doc.rust-lang.org/1.33.0/book/ch18-03-pattern-syntax.html#legacy-patterns-ref-and-ref-mut) and can be confusing to newer Rust users.